pkg/mutation/mutation_function.go

```diff
@@ -0,0 +1,140 @@
+package mutation
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func Mutate(mutator Mutator, obj *unstructured.Unstructured) error {
+	return mutate(mutator, obj.Object, nil, 0)
+}
+
+func mutate(m Mutator, current interface{}, previous interface{}, depth int) error {
+	if len(m.Path().Nodes)-1 == depth {
+		return addValue(m, current, previous, depth)
+	}
+	pathEntry := m.Path().Nodes[depth]
+	switch t := pathEntry.Type(); t {
+	case parser.ObjectNode:
+		next, ok := current.(map[string]interface{})[pathEntry.(*parser.Object).Reference]
+		if !ok {
+			next = createMissingElement(m, current, previous, depth)
+		}
+		if err := mutate(m, next, current, depth+1); err != nil {
+			return err
+		}
+		return nil
+	case parser.ListNode:
+		elementFound := false
+		listPathEntry := pathEntry.(*parser.List)
+		glob := listPathEntry.Glob
+		key := listPathEntry.KeyField
+		for _, listElement := range current.([]interface{}) {
+			if glob {
+				if err := mutate(m, listElement, current, depth+1); err != nil {
+					return err
+				}
+				elementFound = true
+			} else if elementValue, ok := listElement.(map[string]interface{})[key]; ok {
```

Comment:
needs safe type-casting

Reply:
Done

Reply:
Where is the safe type conversion?

Reply:
RIght! The "ok" is for the key lookup, not casting!
Sorry for this. Fixed now.

We will skip all element which can not be cast to a map. This would only be the case if we have a slice of slices (don't think our path entries cover this case)